### PR TITLE
Re-enable NetFX in FileSystem tests

### LINKED ISSUE #31256: Add back netfx test run in System.IO.FileSystem
Per https://github.com/dotnet/corefx/pull/41452#issuecomment-544650139

If instead we decide to close this issue, please delete the netfx-specific tests added in the above PR.

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes https://github.com/dotnet/runtime/issues/31256

Some files had to be split because the features they test are only available in .NET Core.

### CODE REVIEW

File: src/libraries/System.IO.FileSystem/tests/File/Delete.cs [hunk 1]
```diff
@@ -186,10 +186,10 @@ public void UnixDeleteReadOnlyFile()
             Assert.False(testFile.Exists);
         }
 
-        [Theory,
-            InlineData(":bar"),
-            InlineData(":bar:$DATA")]
         [PlatformSpecific(TestPlatforms.Windows)]
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNetCore))]
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/tests/File/Delete.cs]: We generally instead use:
```C#
[Theory]
[SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "explanation")]
```

File: src/libraries/System.IO.FileSystem/tests/File/Move.netcoreapp.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class File_Move : FileSystemTest
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/tests/File/Move.netcoreapp.cs]: I'm assuming all of the code that was moved between files wasn't modified at all and thus I don't need to review it, correct?

**Comment by stephentoub** [comment]
There are 852 test failures in the System.IO.FileSystem.Tests on .NET Framework ;)

File: src/libraries/System.IO.FileSystem/tests/File/Move.netcoreapp.cs (hunk 2 above)
Comment (carlossanlop) [reply src/libraries/System.IO.FileSystem/tests/File/Move.netcoreapp.cs]: Correct, no change aside from the "partial" keyword.

**Comment by carlossanlop** [comment]
>  _There are 852 test failures in the System.IO.FileSystem.Tests on .NET Framework ;)_

Strange. I fixed the ones I hit locally (they only were features unavailable in Framework). I didn't hit any regressions. I'll investigate.

**Comment by carlossanlop** [comment]
This will take a while to address. I'll close the PR for now.

**Comment by stephentoub** [comment]
Thanks

**Comment by carlossanlop** [comment]
Force pushing prevents me from reopening this PR. I'll have to submit a new one.

**Pull request closed without merging**